wkcuber/api/Layer.py

```diff
@@ -353,17 +415,23 @@ def setup_mag(self, mag: Union[str, Mag]) -> None:
 
         self._assert_mag_does_not_exist_yet(mag)
 
-        with wkw.Dataset.open(
-            find_mag_path_on_disk(self.dataset.path, self.name, mag)
-        ) as wkw_dataset:
-            wk_header = wkw_dataset.header
+        try:
+            with wkw.Dataset.open(
+                find_mag_path_on_disk(self.dataset.path, self.name, mag)
+            ) as wkw_dataset:
+                wk_header = wkw_dataset.header
 
-        self.mags[mag] = WKMagDataset(
-            self, mag, wk_header.block_len, wk_header.file_len, wk_header.block_type
-        )
-        self.dataset.properties._add_mag(
-            self.name, mag, cube_length=wk_header.block_len * wk_header.file_len
-        )
+            self.mags[mag] = WKMagDataset(
+                self, mag, wk_header.block_len, wk_header.file_len, wk_header.block_type
+            )
+
+            self.dataset.properties._add_mag(
+                self.name, mag, cube_length=wk_header.block_len * wk_header.file_len
+            )
```

Comment:
Leaving this inside the `try`-block means that it will eventually remove the entry from the `datasource-properties.json`. Do you think this is okay?

Reply:
I don't understand yet. In what cases would the entry be removed?

Reply:
When opening an existing dataset, the metadata is stored in an `Properties` object. These properties are used to write the `datasource-properties.json` to disk if necessary. 
If an `WKDataset` is opened, which has e.g. an entry for `Mag(2)` but this mag does not exist on disk (this might be the case if the directory is deleted, but the properties are not updated), this implementation  would fail in the `try` block. Therefore, the entry would not be added to the properties object in RAM. The next time the properties are dumped to disk (e.g. because the bounding box changed), the file would be overwritten (this would "remove" the previously existing entry of `Mag(2)` from the properties on disk)

Reply:
Yeah. I think that is ok.